site/zenodo_rdm/legacy/deserializers/schemas.py

```diff
@@ -261,10 +463,76 @@ def _metadata(self, data):
                 "rights": self._rights(data.get("license")),
                 "contributors": self._contributors(data),
                 "additional_descriptions": self._additional_descriptions(
-                    data.get("notes")
+                    {"notes": data.get("notes"), "method": data.get("method")}
                 ),
+                "locations": self._locations(data.get("locations")),
+                "subjects": self._subjects(data),
+                "version": data.get("version"),
+                "dates": self._dates(data.get("dates", [])),
+                "references": self._references(data.get("references", [])),
+                "languages": self._language(data.get("language")),
+                "related_identifiers": self._related_identifiers(
+                    data.get("related_identifiers", [])
+                ),
+            }
+
+    def _access(self, data):
+        """Transform the access of a record."""
+
+        def _embargoed_record(embargo_date):
+            """Generates RDM access conditions based on legacy embargo conditions."""
+            return {
+                "record": "public",
+                "files": "restricted",
+                "embargo": {
+                    "active": True,  # boolean
+                    "until": embargo_date,  # ISO String
+                },
+            }
+
+        def _restricted_record():
+            """Generates RDM access conditions based on legacy restricted conditions."""
+            return {
+                "record": "public",
+                "files": "restricted",
             }
 
+        def _open_record():
+            """Generates RDM access conditions for an open record."""
+            return {
+                "record": "public",
+                "files": "public",
+            }
+
+        default = _open_record()
+        access = None
+        metadata = data.get("metadata", {})
+        access_right = metadata.get("access_right")
+        embargo_date = metadata.get("embargo_date")
+
+        if not access_right:
+            # TODO throw an error actually? Or default to public?
```

Comment:
This is more of a validation check. Access right should be there always, so I think should be tackled in a separate [issue](https://github.com/zenodo/zenodo-rdm/issues/334) when we defined a strategy for error handling.

Reply:
Wouldnt it be safer to translate to closed? to avoid data leaks of datasets? i.e. user makes a mistake and sends unintendedly open, but should have been closed.

I guess is our "part" on promoting open data, the user can always go an edit. Right?

Reply:
I think both have good points.

Also, the same can happen when uploading through the UI (e.g. user forgets about the access rights). In that case, the UI defaults to open access. 

About mistakes: yeah that would be a pain for the user, especially if they're uploading a bunch of records using a script. I think they would have to change it in the UI, one by one.